src/get-env-variables.js

```diff
@@ -0,0 +1,60 @@
+const exec = require('child_process').exec
+// import spawn from 'spawn-command-with-kill'
+import {assign, clone} from 'lodash'
+import managePath from 'manage-path'
+import {sync as findUpSync} from 'find-up'
+
+const NON_ERROR = 0
+
+export default getEnv
+
+function getEnv() {
+  return new Promise((resolve, reject) => {
+    getNpmEnvVariables()
+      .then(npmVars => {
+        const env = assign(clone(process.env), npmVars)
+        const alterPath = managePath(env)
+        const npmBin = findUpSync('node_modules/.bin')
+        if (npmBin) {
+          alterPath.unshift(npmBin)
+        }
+        resolve(env)
+      })
+      .catch(reject)
+  })
+}
+
+function getNpmEnvVariables() {
+  let child
+  return new Promise((resolve, reject) => {
+    let output = ''
+    child = exec('npm run env')
```

Comment:
Don't know if we can avoid having to spawn anything.

The only promising thing that I could find was this: https://github.com/npm/npm/blob/d081cc6c8d73f2aa698aab36605377c95e916224/lib/config.js#L188. But this would require us to have npm as a dependency, which doesn't sound like a better option to me.